pkg/install/install.go

```diff
@@ -50,20 +48,28 @@ import (
 	"github.com/Azure/ARO-RP/pkg/util/privateendpoint"
 	"github.com/Azure/ARO-RP/pkg/util/refreshable"
 	"github.com/Azure/ARO-RP/pkg/util/restconfig"
+	"github.com/Azure/ARO-RP/pkg/util/steps"
 	"github.com/Azure/ARO-RP/pkg/util/stringutils"
 	"github.com/Azure/ARO-RP/pkg/util/subnet"
 	"github.com/Azure/ARO-RP/pkg/util/version"
 )
 
 // Installer contains information needed to install an ARO cluster
 type Installer struct {
-	log          *logrus.Entry
-	env          env.Interface
-	db           database.OpenShiftClusters
-	billing      billing.Manager
-	doc          *api.OpenShiftClusterDocument
-	cipher       encryption.Cipher
-	fpAuthorizer refreshable.Authorizer
+	log               *logrus.Entry
+	env               env.Interface
+	db                database.OpenShiftClusters
+	billing           billing.Manager
+	doc               *api.OpenShiftClusterDocument
+	subscriptiondoc   *api.SubscriptionDocument
+	cipher            encryption.Cipher
+	fpAuthorizer      refreshable.Authorizer
+	localFPAuthorizer refreshable.Authorizer
```

Comment:
I can't tell the difference between `fpAuthorizer` and `localFPAuthorizer` very easily - is it possible to choose clearer names that indicate their use or add a comment?

Reply:
Adding my 2c as I did a little digging around this area recently.

We use these names in multiple places. Local means RP's tenant (it is used to access DNS and Private Endpoint resources) and the `fpAuthorizer` is used for other resources in cluster RG such as VM, SG, LB, etc: 
https://github.com/Azure/ARO-RP/blob/a06bf3d2241501c532d36d207356cc4c1303b672/pkg/install/install.go#L128-L150

For `localFPAuthorizer` we use instance metadata service: https://github.com/Azure/ARO-RP/blob/master/pkg/util/instancemetadata/prod.go (for dev we simply populate these values from env vars)

For `fpAuthorizer` we use `TenantID` from customer's subscription document, which is stored in DB and updated by ARM. 

Some more information about different identities that we use can be found here: https://github.com/Azure/ARO-RP/blob/master/docs/prepare-a-shared-rp-development-environment.md#aad-applications

Reply:
I agree that this might be not intuitive, but I suggest that we don't change these names in scope of this PR as it will be inconsistent with other places where we use the same approach, e.g: https://github.com/Azure/ARO-RP/blob/a06bf3d2241501c532d36d207356cc4c1303b672/pkg/backend/openshiftcluster/openshiftcluster.go#L47

Reply:
@olga-mir That's a good explanation. In lieu of renaming these fields, then, perhaps a comment or two explaining (at least partially) the above would be in order.

Reply:
@olga-mir, spot on.  @nilsanderselde please can you take the action to find somewhere appropriate to document this, and document it? (move the discussion away from this PR)